packages/wdio-cucumber-framework/src/utils.js

```diff
@@ -92,37 +97,44 @@ export function formatMessage ({ type, payload = {} }) {
         type: type
     }
 
-    if (payload.error) {
-        message.error = payload.error
-
-        /**
-         * hook failures are emitted as "test:fail"
-         */
-        if (payload.title && payload.title.match(/^"(before|after)( all)*" hook/g)) {
-            message.type = 'hook:end'
-        }
-    }
-
-    /**
-     * Add the current test title to the payload for cases where it helps to
-     * identify the test, e.g. when running inside a beforeEach hook
-     */
-    if (payload.ctx && payload.ctx.currentTest) {
-        message.currentTest = payload.ctx.currentTest.title
-    }
-
-    if (type.match(/Test/)) {
-        message.passed = (payload.state === 'passed')
-    }
-
     if (payload.title && payload.parent) {
         message.fullTitle = getTestFullTitle(payload.parent, payload.title)
     }
 
     return message
 }
 
-export function compareScenarioLineWithSourceLine (scenario, sourceLocation) {
+/**
+ * Get step type
+ * @param {string} type `Step` or `Hook`
+ */
+export function getStepType(type) {
+    return type === 'Step' ? 'test' : 'hook'
+}
+
+/**
+ * build payload for test/hook event
+ */
+export function buildStepPayload(uri, feature, scenario, step, type, title, state, error, duration, passed) {
```

Comment:
Is there a way to limit the amount of args and provide an object?

Reply:
Yeah, I was also thinking to do this and merge object instead

Reply:
@wswebcreation done